### ISSUE TITLE
`design` vs `whitened_design` when computing `predicted` in a `glm`

### ISSUE THREAD

**Issue by mathias-sm** [nilearn/nilearn#2964]
Dear nilean team,

After finding unexpected results when computing a `predicted` signal myself, I noticed that `SimpleRegressionResults` and `RegressionResults` do not compute the `predicted` signal the same way: the former uses `self.model.design` and the other `self.whitened_design` — although they both use `self.theta`.

https://github.com/nilearn/nilearn/blob/e6591d8907ceae55eea6f04ce181acb6a12743b5/nilearn/glm/regression.py#L543-L549

https://github.com/nilearn/nilearn/blob/e6591d8907ceae55eea6f04ce181acb6a12743b5/nilearn/glm/regression.py#L427-L433

I do not know if this is a bug or not and could not find any documentation detailing this difference, but although I noticed that the betas are computed whitened-`design`/`Y`. Would it be possible to detail the difference, if meaningful? If not, adding a test checking that the `.predicted` of either are the same could catch future divergence — I could work on adding that provided it is deemed useful by the nilearn team.

Hope this helps,

Many thanks,

**Comment by bthirion** [comment]
Oops. Normally the right one is self.model.design.

**Comment by mathias-sm** [comment]
Oh, I assumed otherwise in the meantime because of the way `beta`s are computed

https://github.com/nilearn/nilearn/blob/e6591d8907ceae55eea6f04ce181acb6a12743b5/nilearn/glm/regression.py#L248-L249

Where `calc_beta` is computed from `whitened_design`

https://github.com/nilearn/nilearn/blob/e6591d8907ceae55eea6f04ce181acb6a12743b5/nilearn/glm/regression.py#L117

**Comment by bthirion** [comment]
Indeed, you need to whiten the design to estimate the beta through OLS and obtain the mininum variance estimator, but if you want yo get the actual residual, you need to get back to the original model.
A good way to check that is to check with simulated data, where the residuals would be known.

**Comment by mathias-sm** [comment]
Thanks, if I find the time I'll try on synthetic data, see what's what, and report back.

--------

For future reference and to avoid an xy problem, I can detail what I was trying to do briefly. For a plot I had in mind, I generated (i) the activation using the predictors of interest (some betas times the design matrix) and (ii) the rest of the signal using "uninteresting" factors (movements, constant and drifts, etc., the other betas times the design matrix). Then I plotted (i) against the raw signal minus (ii), but the time series didn't overlap in systematic ways so I started digging and ended up comparing my reconstructed predicted signal to that of the glm with `minimize_memory=false` and realized something was off.
